contracts/Controller.sol

```diff
@@ -124,38 +130,89 @@ contract Controller is Ownable {
     /**
      * @notice Return a vault balances, depend of the short option expiry
      * @dev if vault have no short option or issued option is not expired yet, return vault, else get excess margin and return it as collateral amount inside Vault struct.
-     * @param _owner vault owner.
+     * @param _owner account owner.
      * @param _vaultId vault.
      * @return Vault struct
      */
-    // function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
-    // }
+    function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
+        MarginAccount.Vault memory vault = getVault(_owner, _vaultId);
+
+        if ((vault.shortOtokens.length != 0) && (isExpired(vault.shortOtokens[0]))) {
+            address calculatorModule = AddressBookInterface(addressBook).getMarginCalculator();
+            MarginCalculatorInterface calculator = MarginCalculatorInterface(calculatorModule);
+
+            (uint256 netValue, ) = calculator.getExcessMargin(vault, vault.shortOtokens[0]);
+            vault.collateralAmounts[0] = netValue;
+
+            return vault;
+        }
+
+        return vault;
```

Comment:
```suggestion

        if ((vault.shortOtokens.length == 0) || (!isExpired(vault.shortOtokens[0]))) 
            return vault;
        
        // if there's short and it's expired 
        address calculatorModule = AddressBookInterface(addressBook).getMarginCalculator();
        MarginCalculatorInterface calculator = MarginCalculatorInterface(calculatorModule);
       (uint256 netValue, ) = calculator.getExcessMargin(vault, vault.shortOtokens[0]);
        vault.collateralAmounts[0] = netValue;

       return vault;
        }        
```
Personally prefer guard clauses.

Reply:
Any reason for that ?

Reply:
just a normal pattern to avoid using big if-else blocks.

Reply:
Simple reason is that when people read til Line 150, they will already forget what the vault is for. If we just need to return it, better do it first, than deal with another logic